pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -4192,50 +4236,103 @@ def PumpCostRule(b, l1, l2):
             units=pyunits.meter,
             doc="Hazen-Williams Frictional loss, m",
         )
-        mh.v_effective_Pipeline_diameter = Var(
-            model.s_LLA,
-            initialize=0,
-            units=model.model_units["diameter"],
-            doc="Diameter of pipeline between two locations [inch]",
-        )
 
-        def EffectiveDiameterRule(b, l1, l2, t1):
-            constraint = mh.v_effective_Pipeline_diameter[
-                l1, l2
-            ] == mh.p_Initial_Pipeline_Diameter[l1, l2] + sum(
-                model.vb_y_Pipeline[l1, l2, d]
-                * model.df_parameters["PipelineDiameterValues"][d]
-                for d in model.s_D
+        # -----------------Removing effective diameter rule. We will put the expression of binaries where needed
```

Comment:
Please avoid overwriting the existing nonlinear hydraulic method. To maintain the integrity of the original work and facilitate meaningful comparisons of outcomes, I suggest creating a separate configuration option for hydraulic calculations with the new linearization approach.

Reply:
Thanks for the comment! I have created an additional configuration option called co_optimize_linearized.